src/components/App/UserContext.tsx

```diff
@@ -0,0 +1,31 @@
+import React from "react"
+
+export interface LoggedInUser {
+  jwt: string
+}
+
+interface LoggedOutUser {
+  jwt: null
+}
+
+export type User = LoggedInUser | LoggedOutUser
+
+interface UserContext {
+  user: User
+  setUser: (u: User) => void
+  logout: () => void
+}
+
+export const AnonymousUser: LoggedOutUser = {
+  jwt: null,
+}
+
+// function isAnonymousUser(u: User): u is AnonymousUser {
+//   return (<Fish>pet).swim !== undefined
+// }
+
+export const UserContext = React.createContext<UserContext>({
+  user: AnonymousUser,
+  setUser: (u: User) => {},
+  logout: () => {},
+})
```

Comment:
There we go 😎

Reply:
Also, would suggest keeping system-level components in the root, under `src/UserContext`. Pieces like this inform the entire app design over time, so good to keep them nice and visible, rather than buried in folders.

Reply:
yeah premature code sorting on my part